### PR TITLE
modified: extracting logic for Kinetica to include partitions

### PR DESCRIPTION
**Pull request by jetroni**
modified: ddl generator to generate partitions creation
modified: AbstractModel to avoid NPE and empty properties

### CODE REVIEW

**Comment by jetroni** [comment]
@nbesimi
The reason why I have changed AbstractModel is to have a better way of adding an additional property without writing it to model.yaml if it's empty.

File: ddl/src/main/java/com/adaptivescale/rosetta/ddl/targets/kinetica/KineticaDDLGenerator.java [hunk 1]
```diff
@@ -85,6 +85,10 @@ public String createTable(Table table, boolean dropTableIfExists) {
         }
 
         String tableType = extractTableType(table);
+        String partitions = table.getPropertyAsString("partitions");
+        if (partitions != null) {
```
Comment (ailegion) [inline ddl/src/main/java/com/adaptivescale/rosetta/ddl/targets/kinetica/KineticaDDLGenerator.java:89]: suggestion: using hash maps containsKey method should be safer alternative

Comment (jetroni) [reply ddl/src/main/java/com/adaptivescale/rosetta/ddl/targets/kinetica/KineticaDDLGenerator.java]: I have modified those parts based on your suggestion.

File: source/src/main/java/com/adataptivescale/rosetta/source/core/extractors/table/KineticaTablesExtractor.java [hunk 2]
```diff
@@ -20,28 +26,57 @@ public class KineticaTablesExtractor extends DefaultTablesExtractor {
...
+                String query = String.format(queryTemplate, table.getSchema(), table.getName());
+                ResultSet resultSet = statement.executeQuery(query);
+                List<Map<String, Object>> records = QueryHelper.mapRecords(resultSet);
+
+                if (!records.isEmpty()) {
+                    table.addProperty("partitions", extractPartitions(records.get(0).get("ddl").toString()));
+                }
+
+            } catch (SQLException e) {
+                log.warn("Skipping processing views due to error: {}", e.getMessage());
```
Comment (nbesimi) [inline source/src/main/java/com/adataptivescale/rosetta/source/core/extractors/table/KineticaTablesExtractor.java:65]: Do we need to say here `Skipping processing views ...`

Comment (jetroni) [reply source/src/main/java/com/adataptivescale/rosetta/source/core/extractors/table/KineticaTablesExtractor.java]: No, I forgot to modify the warn message, now it should be good.

**Review by ailegion — APPROVED** [review]
lgtm

**Pull request merged**